### PR TITLE
Job Scheduling algorithm - Greedy Problem in JAVA

### PR DESCRIPTION
**Pull request by krishna-y2000**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

I have contributed Job Scheduling algorithm in JAVA.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2494

### CODE REVIEW

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
+/* Problem Description
+Given a set of N jobs where each job i has a deadline and profit associated to it. 
+Each job takes 1 unit of time to complete and only one job can be scheduled at a time.
+We earn the profit if and only if the job is completed by its deadline.
+The task is to find the maximum profit and the number of jobs done.
+*/
+//Solution :
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java]: Remove the redundant comments and add a new line after description ends */

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import java.io.*;
+import java.lang.*;
+import java.util.*;
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java:10]: Add a new line after header ends.

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    // this Job class will store the id, profit, deadline property 
+	static class Job {
+	    int id, profit, deadline;
+	    Job(int id, int deadline, int profit){
+            this.id = id;
+            this.profit = profit;
+            this.deadline = deadline;
+        }
+	};
+	// solve() will do the task of getting the profit
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java:23]: Add a new line above the comments

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	static int[] solve (Job[] arr , int n )
+	{
+		// Sort the array in descending order according to profit
+		Arrays.sort(arr , (a,b) -> b.profit - a.profit );
+        int res[]  = new int[2] ;
+        int totalProfit = 0;
+        int count = 0 ;        
+        Boolean[] slots = new Boolean[n];
+        Arrays.fill(slots, false);
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java]: Align your code properly.

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+       /* iterate through the jobs with minDeadLine to remaining jobs and
+         check if there is any slote available and also claculate the profit
+         */
+        for (int i = 0; i < n; i++) {
+        	int minDeadLine = Math.min( arr[i].deadline - 1 , n- 1) ;
+            for (int j = minDeadLine ; j >= 0; j--) 
+            {
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java]: Align your code properly.

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                    break;
+                }
+            }
+        }
+        // store the count and profit in res array and return the result
+        res[0] = count ;
+        res[1] = totalProfit ;
+        return res ;
+	}
+ 	public static void main(String[] args) throws IOException{
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java:54]: Add a new line above the public static

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            for(int i=0, k=0; i<n; i++){
+                arr[i] = new Job(Integer.parseInt(input[k++]), Integer.parseInt(input[k++]), Integer.parseInt(input[k++]));
+            }
+            // call the method to get the result
+            int[] result = solve(arr, n);
+            // Print the required result
+            System.out.println ("Maximum Profit : " + result[0] + " " + result[1]);            
+        }
+    }
+}
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java:76]: Add a new line after the `}`

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+/* Input
+Enter number of testcase
+1
+Enter number of jobs
+4
+Enter values of jobs
+1 4 20 2 1 10 3 1 40 4 1 30
+Maximum Profit : 2 60
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java]: Add a proper new line and space to differentiate b/w Input and output it is bit confusing where input ends and output starts

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+Explanation
+Sort all the jobs according to descending in terms of profit and loop the jobs and check if slot is available or not
+and according insert value to res array and output the result.
+Time Complexity : O(NLogN)
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java:93]: Add a new line above the TC

File: Java/Greedy_Algorithms/Job_Scheduling.java [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+Space Complexity : O(N)
+*/
```
Comment (ankitaggarwal23) [inline Java/Greedy_Algorithms/Job_Scheduling.java:95]: Add a new line at the EOF

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by krishna-y2000** [comment]
@ankitaggarwal23  changes are applied

**Comment by krishna-y2000** [comment]
@ankitaggarwal23  hii,
 if you find any spare time then please review my pr. This might be the last check. 
@HarshCasper

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**